packages/react-html/src/components/HtmlUpdater.tsx

```diff
@@ -43,59 +43,46 @@ function updateOnClient(state: State) {
 
   const fragment = document.createDocumentFragment();
 
-  const oldMetas = Array.from(
-    document.head.querySelectorAll(`meta[${MANAGED_ATTRIBUTE}]`),
-  );
-
-  for (const meta of metas) {
-    const element = document.createElement('meta');
-    element.setAttribute(MANAGED_ATTRIBUTE, 'true');
-
-    for (const [attribute, value] of Object.entries(meta)) {
-      element.setAttribute(attribute, value);
-    }
+  updateElement('meta', metas, fragment);
+  updateElement('link', links, fragment);
+  updateElement('style', styles, fragment);
 
-    const matchingOldMetaIndex = oldMetas.findIndex(oldMeta =>
-      oldMeta.isEqualNode(element),
-    );
-
-    if (matchingOldMetaIndex >= 0) {
-      oldMetas.splice(matchingOldMetaIndex, 1);
-    } else {
-      fragment.appendChild(element);
-    }
-  }
+  document.head.appendChild(fragment);
+}
 
-  const oldLinks = Array.from(
-    document.head.querySelectorAll(`link[${MANAGED_ATTRIBUTE}]`),
+function updateElement<T extends HTMLProps<HTMLElement>>(
+  selector: string,
+  items: T[],
+  fragment: DocumentFragment,
+) {
+  const oldElements = Array.from(
+    document.head.querySelectorAll(`${selector}[${MANAGED_ATTRIBUTE}]`),
   );
 
-  for (const link of links) {
-    const element = document.createElement('link');
+  for (const item of items) {
+    const element = document.createElement(selector);
     element.setAttribute(MANAGED_ATTRIBUTE, 'true');
 
-    for (const [attribute, value] of Object.entries(link)) {
-      element.setAttribute(attribute, value);
+    for (const [attribute, value] of Object.entries(item)) {
+      if (attribute === 'children') {
```

Comment:
This is the only part I changed. Instead of calling `setAttribute` when you pass children, we set the value to the `innerHTML`.

Any XSS issue with that 🤔 ?

Reply:
Decided to use `innerText` instead.